zowe/vscode-extension-for-zowe
# build(api): Enhance watch task to trigger for zowe-explorer-api changes

**What It Does**
Updates our tsconfig and webpack config so that when the watch task is running, changes made to "zowe-explorer-api" package will trigger a new incremental build, and a complete rebuild of the project is no longer required.

Also fixes webpack config to load source maps correctly so that breakpoints can be set within "zowe-explorer-api" code.

See [here](https://medium.com/@nickexcell/using-typescript-project-references-with-ts-loader-and-webpack-part-1-5d0c3cd7c603) for more details about how TS project references can be used with Webpack.

**How to Test**
* Launch the debug task to run ZE or the FTP extension which starts a watch task in the terminal
* Make a change to source code inside the "zowe-explorer-api" package and save the file
* The watch task should automatically trigger to rebuild with code changes made to ZE API
* Set a breakpoint in a TS file in the "zowe-explorer-api" package and make sure it triggers
